internal/services/codesigning/trusted_signing_certificate_profile_resource.go

```diff
@@ -0,0 +1,478 @@
+	Reason        string                               `tfschema:"reason"`
+	Remarks       string                               `tfschema:"remarks"`
+	RequestedAt   string                               `tfschema:"requested_at"`
+	Status        certificateprofiles.RevocationStatus `tfschema:"status"`
+}
+
+type TrustedSigningCertificateProfileResource struct{}
+
+var _ sdk.ResourceWithUpdate = TrustedSigningCertificateProfileResource{}
+
+func (r TrustedSigningCertificateProfileResource) ResourceType() string {
+	return "azurerm_trusted_signing_certificate_profile"
+}
+
+func (r TrustedSigningCertificateProfileResource) ModelObject() interface{} {
+	return &TrustedSigningCertificateProfileModel{}
+}
+
+func (r TrustedSigningCertificateProfileResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return certificateprofiles.ValidateCertificateProfileID
+}
+
+func (r TrustedSigningCertificateProfileResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(5, 100),
+				validation.StringMatch(
+					regexp.MustCompile("^[A-Za-z][A-Za-z0-9]*(?:-[A-Za-z0-9]+)*$"),
+					"A certificate profile's name must be between 3-24 alphanumeric characters. The name must begin with a letter, end with a letter or digit, and not contain consecutive hyphens.",
+				)),
+		},
+
+		"trusted_signing_account_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: codesigningaccounts.ValidateCodeSigningAccountID,
+		},
+
+		"identity_validation_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"profile_type": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(certificateprofiles.ProfileTypePublicTrust),
+				string(certificateprofiles.ProfileTypePrivateTrust),
+				string(certificateprofiles.ProfileTypePrivateTrustCIPolicy),
+				string(certificateprofiles.ProfileTypeVBSEnclave),
+				string(certificateprofiles.ProfileTypePublicTrustTest),
+			}, false),
+		},
+
+		"include_city": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"include_country": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"include_postal_code": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"include_state": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"include_street_address": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+	}
+}
+
+func (r TrustedSigningCertificateProfileResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"certificates": {
```

Comment:
Most of the attribute here seems need to be removed as they are meaningless for TF users?

Reply:
not sure about this..

Reply:
removed